packages/ampd-sdk/src/grpc/client.rs

```diff
@@ -56,11 +71,41 @@ pub async fn new(url: &str) -> Result<GrpcClient, Error> {
 #[async_trait]
 #[allow(clippy::todo)]
 impl Client for GrpcClient {
+    type Stream = Pin<Box<dyn Stream<Item = Result<Event, Error>> + Send>>;
+
     async fn subscribe(
-        &self,
-        _request: SubscribeRequest,
-    ) -> Result<Streaming<SubscribeResponse>, Error> {
-        todo!()
+        &mut self,
+        filters: Vec<AbciEventTypeFilter>,
+        include_block_begin_end: bool,
+    ) -> Result<Self::Stream, Error> {
+        let request = SubscribeRequest {
+            filters: filters
+                .into_iter()
+                .map(|filter| ampd_proto::Event {
+                    r#type: filter,
+                    contract: String::new(),
+                    attributes: HashMap::new(),
+                })
+                .collect(),
+            include_block_begin_end,
+        };
+
+        let streaming_response = self
+            .blockchain
+            .subscribe(request)
+            .await
+            .map_err(Error::GrpcRequest)
+            .map_err(Report::new)?;
+
+        let transformed_stream = streaming_response.into_inner().map(|result| match result {
+            Ok(response) => match response.event {
+                Some(event) => Event::try_from(event).map_err(|_| report!(Error::EventConversion)),
```

Comment:
```suggestion
                Some(event) => Event::try_from(event).change_context(Error::EventConversion),
```
so that the inner error isn't swallowed.